### PR TITLE
Add milliseconds per move timing info

### PR DESCRIPTION
**Pull request by sethtroisi**
https://i.imgur.com/X0aQV68.png

### CODE REVIEW

**Comment by kingvash** [comment]
Can this be pulled? It's useful in testing performance (where number of games is way to coarse)

File: autogtp/Game.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ class Game : QProcess {
     bool waitForMove() { return waitReady(); }
     bool readMove();
     bool nextMove();
+    int  getMoveNum();
```
Comment (gcp) [inline autogtp/Game.h]: nit: there's an extra space here

File: autogtp/Production.cpp [hunk 2]
```diff
@@ -37,16 +37,18 @@ void ProductionWorker::run() {
              game.readMove();
          } while (game.nextMove());
          QTextStream(stdout) << "Game has ended." << endl;
+         int movesMade = 1;
```
Comment (gcp) [inline autogtp/Production.cpp]: The way this value is set as the default makes me suspicious the error handling around here is wonky. Notably, should we call emit resultReady when we can't do a getScore()?  Ignoring that problem (which exists in the current code), can't we already just call game.getMoveNum() here?

**Comment by sethtroisi** [comment]
Changed calculation slightly, now calculates time / all_moves (from finished and partial games)

this make the output much more stable (instead of bouncing around passed on how many outstanding moves had been calculated but where part of games that hadn't finished), and more useful as a real performance evaluation (as compared to the single instance netbench)

1 game(s) played in 1 minutes = 95 seconds/game, 372 ms/move, last game took 94 seconds.
12 game(s) played in 14 minutes = 73 seconds/game, 381 ms/move, last game took 163 seconds.
22 game(s) played in 28 minutes = 78 seconds/game, 396 ms/move, last game took 135 seconds.
36 game(s) played in 47 minutes = 78 seconds/game, 390 ms/move, last game took 114 seconds.

File: autogtp/Production.cpp [hunk 3]
```diff
@@ -178,16 +182,22 @@ void Production::getResult(const QString& file, float duration) {
...
+    if (m_movesMade == 0 || m_gamesPlayed == 0) {
+        return;
+    }
     auto game_end = std::chrono::high_resolution_clock::now();
     auto total_time_s =
         std::chrono::duration_cast<std::chrono::seconds>(game_end - m_start);
     auto total_time_min =
         std::chrono::duration_cast<std::chrono::minutes>(total_time_s);
+    auto total_time_millis =
+        std::chrono::duration_cast<std::chrono::milliseconds>(total_time_s);
```
Comment (sethtroisi) [inline autogtp/Production.cpp]: I should really use (game_end - m_start) to calculate total_time_millis then cast that to total_time_s and total_time_min but the net effect is in the 4 or 5th sig fig.

**Pull request merged**